### ISSUE TITLE
Neural Moving Horizon Estimation for Robust Flight Control. (arXiv:2206.10397v9 [cs.RO] UPDATED)

### ISSUE THREAD

**Issue by CoffeeKumazaki** [CoffeeKumazaki/arXiv#20845]
Neural Moving Horizon Estimation for Robust Flight Control. (arXiv:2206.10397v9 [cs.RO] UPDATED)<br>
https://ift.tt/bBmlXaR<br>
<p>Estimating and reacting to external disturbances is crucial for robust flight control of quadrotors. Existing estimators typically require significant tuning for a specific flight scenario or training with extensive ground-truth disturbance data to achieve satisfactory performance. In this paper, we propose a neural moving horizon estimator (NeuroMHE) that can automatically tune the key parameters modeled by a neural network and adapt to different flight scenarios. We achieve this by deriving the analytical gradients of the MHE estimates with respect to the weighting matrices, which enables a seamless embedding of the MHE as a learnable layer into neural networks for highly effective learning. Interestingly, we show that the gradients can be computed efficiently using a Kalman filter in a recursive form. Moreover, we develop a model-based policy gradient algorithm to train NeuroMHE directly from the quadrotor trajectory tracking error without needing the ground-truth disturbance data. The effectiveness of NeuroMHE is verified extensively via both simulations and physical experiments on quadrotors in various challenging flights. Notably, NeuroMHE outperforms the state-of-the-art neural network-based estimator with estimation error reductions of up to about 49.4% by using only a 2.5% amount of the neural network parameters. The proposed method is general and can be applied to robust adaptive control of other robotic systems.</p>
<br>
<br>
via cs.RO updates on arXiv.org http://arxiv.org/<br>
